API/DTOs/SearchResponseCollectionDTO.cs

```diff
@@ -0,0 +1,22 @@
+namespace API.DTOs;
+
+/// <summary>
+/// Represents a collection of search results
+/// </summary>
+public class SearchResponseCollectionDTO
+{
+    /// <summary>
+    /// A collection of search results for tournaments matching the search query
+    /// </summary>
+    public SearchResponseDTO[] Tournaments { get; set; } = null!;
```

Comment:
Any reason for using array vs ICollection / IList etc.?

Reply:
> If you know the data is fixed length, and you want to micro-optimise for some very specific reason (after benchmarking), then an array may be useful.

_technically_ for data transfers arrays are marginally more efficient since they are immutable.